deta/base.py

```diff
@@ -100,10 +100,22 @@ def __init__(self, drive_name:str=None, project_key:str=None, project_id:str=Non
         self.project_key = project_key
         self.project_id = project_id
         host = host or os.getenv("DETA_DRIVE_HOST") or "drive.deta.sh"
-        self.client = http.client.HTTPSConnection(host, timeout=3)
+        self.client = http.client.HTTPConnection(host, timeout=300)
         self.base_path = "/v1/{0}/{1}".format(self.project_id, self.name)
         self.util = Util()
 
+    def delete(self, name: str) -> typing.Optional[str]:
+        """Delete an item from drive
+        name: the name of item to be deleted
+        """
+        if name == "":
+            raise ValueError("Name is empty")
+        # encode key
+        key = quote(name, safe="")
+        status, payload = self._request("/files", "DELETE", {"names":[name]})
+        if (status == 404):
+            return None
+        return payload["deleted"]
```

Comment:
delete returns a `2xx` status code even if some deletion fails and the failed files are returned in `payload['failed']`

we need to raise an error if deletion fails here

Reply:
Gotcha, thanks!